### PR TITLE
Feature: add meta grpc client network metrics

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Feature: add meta grpc client network metrics

## Changelog

- New Feature

## Related Issues

Fixes [#issue 5977](https://github.com/datafuselabs/databend/issues/5977)

### CODE REVIEW

File: common/meta/grpc/src/grpc_client.rs [hunk 1]
```diff
@@ -75,6 +78,11 @@ use crate::METACLI_COMMIT_SEMVER;
 use crate::MIN_METASRV_SEMVER;
 
 const AUTH_TOKEN_KEY: &str = "auth-token-bin";
+const META_GRPC_CLIENT_REQUEST_DURATION_MILLIS: &str = "meta_grpc_client_request_duration_millis";
```
Comment (everpcpc) [inline common/meta/grpc/src/grpc_client.rs:81]: `millis` -> `ms` ?

**Review by drmingdrmer — APPROVED** [review]
Nice host!

File: common/meta/grpc/src/grpc_client.rs [hunk 2]
```diff
@@ -321,11 +332,36 @@ impl MetaGrpcClient {
...
+                if success {
+                    label_counter_with_val_and_labels(
+                        META_GRPC_CLIENT_REQUEST_SUCCESS,
+                        vec![(LABEL_ENDPOINT, current_endpoint.to_string())],
+                        1,
+                    );
+                } else {
+                    label_counter_with_val_and_labels(
+                        META_GRPC_CLIENT_REQUEST_FAILED,
+                        vec![(LABEL_ENDPOINT, current_endpoint.to_string())],
```
Comment (flaneur2020) [inline common/meta/grpc/src/grpc_client.rs:361]: can we label the `failed` metric with error kind?

File: common/meta/grpc/src/grpc_client.rs [hunk 3]
```diff
@@ -390,7 +429,10 @@ impl MetaGrpcClient {
                     let ch = self.conn_pool.get(addr).await;
                     match ch {
                         Ok(c) => {
-                            *current_endpoint = Some(addr.clone());
+                            {
+                                let mut current_endpoint = self.current_endpoint.write().await;
+                                *current_endpoint = Some(addr.clone());
```
Comment (drmingdrmer) [inline common/meta/grpc/src/grpc_client.rs:442]: If current_endpoint  does not need to be locked across await, using a std::sync::Mutex would be more efficient.

**Review by drmingdrmer — APPROVED** [review]
Make this PR approved again!

**Review by ariesdevil — APPROVED** [review]

**Pull request merged**